### DISCUSSION TITLE
Is there a way of submitting a menu item using a key that corresponds to an item?

### DISCUSSION THREAD

**Discussion by eddiemundo** [MunifTanjim/nui.nvim#57] [Q&A] [answered]
For instance in coc.nvim I can press the number that matches the menu item to select the item. I can also cycle through them using using the up and down keys as well as j, and k.

It doesn't look like you can do this currently.

I guess a simple way of doing this is allowing a menu item to have a key, then wire up the keymapping from that implicitly. Downside is maybe it's not obvious that that changes the keymapping. Another possible way that is more long winded, is to be able to provide an optional id for each menu item. Then provide an option in the keymap table that is a list of (key, menu item id pairs).

**Comment by MunifTanjim** [comment, marked as answer]
It's not currently built-in for `nui`. But nui is very flexible and you can definitely do it yourself.

For example it's done here: https://github.com/MunifTanjim/exrc.nvim/blob/01c97ddc/lua/exrc/init.lua#L127-L172

You can create an abstraction layer on top of it. For example, you can do something like this:

```lua
  local lines = {
    Menu.item("[A]llow", { action = "allow", key = "a" }),
    Menu.item("[D]isallow", { action = "disallow", key = "d" }),
    Menu.item("[C]lose", { key = "c" }),
    Menu.item("[O]pen", { action = "open", key = "o" }),
  }

 local menu = Menu(popup_options, {
    lines = lines,
    -- other options
  })

  menu:mount()

  for _, line in iparis(lines) do
    menu:map("n", line.key, function()
      menu:unmount()
      if line.action == "allow" then
        -- do something
      else
        -- do something else
      end
    end, map_options)
  end
```

P.S.: you can store anything inside a `Menu.item` and use them however you want.

**Comment by eddiemundo** [comment]
Ah thank you, I didn't realize that you could do the mapping outside of the configuration.
